components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java

```diff
@@ -1721,11 +1729,55 @@ public static APIProduct importApiProduct(String extractedFolderPath, Boolean pr
+                apiProductRevision.setApiUUID(importedAPIUuid);
+                apiProductRevision.setDescription("Revision created after importing the API Product");
+                try {
+                    revisionId = apiProvider.addAPIProductRevision(apiProductRevision);
+                } catch (APIManagementException e) {
+                    //if the revision count is more than 5, addAPIRevision will throw an exception. If rotateRevision
+                    //enabled, earliest revision will be deleted before creating a revision again
+                    if (e.getErrorHandler().getErrorCode() ==
+                            ExceptionCodes.from(ExceptionCodes.MAXIMUM_REVISIONS_REACHED).getErrorCode() &&
+                            rotateRevision) {
+                        String earliestRevisionUuid = apiProvider.getEarliestRevisionUUID(importedAPIUuid);
+                        List<APIRevisionDeployment> deploymentsList =
+                                apiProvider.getAPIRevisionDeploymentList(earliestRevisionUuid);
+                        //if the earliest revision is already deployed in gateway environments, it will be undeployed
+                        //before deleting
+                        apiProvider
+                                .undeployAPIProductRevisionDeployment(importedAPIUuid, earliestRevisionUuid, deploymentsList);
+                        apiProvider.deleteAPIProductRevision(importedAPIUuid, earliestRevisionUuid);
+                        revisionId = apiProvider.addAPIProductRevision(apiProductRevision);
+                    } else {
+                        throw new APIManagementException(e);
+                    }
                 }
-                addClientCertificates(extractedFolderPath, apiProvider, preserveProvider,
-                        importedApiProduct.getId().getProviderName());
+
+                //Once the new revision successfully created, artifacts will be deployed in mentioned gateway
+                //environments
+                List<APIRevisionDeployment> apiProductRevisionDeployments = new ArrayList<>();
+                for (int i = 0; i < deploymentInfoArray.length(); i++) {
+                    APIRevisionDeployment apiProductRevisionDeployment = new APIRevisionDeployment();
+                    apiProductRevisionDeployment.setDeployment(
+                            deploymentInfoArray.getJSONObject(i).getString(ImportExportConstants.DEPLOYMENT_NAME));
+                    apiProductRevisionDeployment
+                            .setDisplayOnDevportal(deploymentInfoArray.getJSONObject(i)
+                                    .getBoolean(ImportExportConstants.DISPLAY_ON_DEVPORTAL_OPTION));
+                    apiProductRevisionDeployments.add(apiProductRevisionDeployment);
+                }
+                apiProvider.addAPIProductRevisionDeployment(importedAPIUuid, revisionId, apiProductRevisionDeployments);
```

Comment:
Can we rename `addAPIProductRevisionDeployment` to `deployAPIProductRevision` for the sake of clarity?

Reply:
These methods are not introduced by me. These were added from the revision feature. Shall I change them as well?

Reply:
Ya just check with @CrowleyRajapakse and change the name if its okay, otherwise its a bit confusing

Reply:
+1 to change

Reply:
Fixed with 977e3d5